tests/config/test_workers.py

```diff
@@ -77,7 +87,13 @@ def test_old_configs_appservice_worker(self) -> None:
         Tests old (legacy) config options. This is for the worker's config.
         """
         appservice_worker_config = self._make_worker_config(
-            worker_app="synapse.app.appservice", worker_name="worker1"
+            worker_app="synapse.app.appservice",
+            worker_name="worker1",
+            extras={
+                # Set notify_appservices to false for the initialiser's config,
+                # so that it doesn't raise an exception here.
+                "notify_appservices": False,
```

Comment:
Does this get read by `_should_this_worker_perform_duty` at all?
A comment explaining that it doesn't would be reassuring.

Same for the instance below.